x/concentrated-liquidity/types/gov_test.go

```diff
@@ -108,3 +108,103 @@ func TestTickSpacingDecreaseProposalMarshalUnmarshal(t *testing.T) {
 		require.Equal(t, *test.proposal, decoded)
 	}
 }
+
+func TestCreateConcentratedLiquidityPoolsProposal_ValidateBasic(t *testing.T) {
+	defaultRecords := []types.PoolRecord{
+		{
+			Denom0:             "uion",
+			Denom1:             "uosmo",
+			TickSpacing:        100,
+			ExponentAtPriceOne: sdk.NewInt(-1),
+			SpreadFactor:       sdk.MustNewDecFromStr("0.01"),
+		},
+		{
+			Denom0:             "stake",
+			Denom1:             "uosmo",
+			TickSpacing:        1000,
+			ExponentAtPriceOne: sdk.NewInt(-5),
+			SpreadFactor:       sdk.MustNewDecFromStr("0.02"),
+		},
+		{
+			Denom0:             "ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2",
+			Denom1:             "uosmo",
+			TickSpacing:        10,
+			ExponentAtPriceOne: sdk.NewInt(-3),
+			SpreadFactor:       sdk.MustNewDecFromStr("0.05"),
+		},
+	}
+
+	tests := []struct {
+		name                string
+		records             []types.PoolRecord
+		invalidTickSpacing  bool
+		invalidSameDenom    bool
+		invalidDenom0       bool
+		invalidDenom1       bool
+		invalidSpreadFactor bool
```

Comment:
I think a with design pattern is generally better, where you apply a function to some base copy, to make a record w/ some edit

Reply:
"with" design pattern implemented here https://github.com/osmosis-labs/osmosis/pull/5548/commits/a567a70e694d62a0cb4a212f53215b1d759e7eb2